app/src/main/java/net/squanchy/navigation/firststart/FirstStartWithNoNetworkActivity.kt

```diff
@@ -0,0 +1,189 @@
+package net.squanchy.navigation.firststart
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.content.Intent
+import android.graphics.drawable.AnimatedVectorDrawable
+import android.graphics.drawable.Drawable
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.support.v4.view.animation.FastOutLinearInInterpolator
+import android.util.Property
+import android.view.View
+import android.view.animation.BounceInterpolator
+import android.view.animation.Interpolator
+import android.view.animation.LinearInterpolator
+import kotlinx.android.synthetic.main.activity_first_start_with_no_network.firstStartCta
+import kotlinx.android.synthetic.main.activity_first_start_with_no_network.firstStartNevermind
+import kotlinx.android.synthetic.main.activity_first_start_with_no_network.firstStartProgress
+import net.squanchy.R
+import net.squanchy.fonts.TypefaceStyleableActivity
+import net.squanchy.support.config.DialogLayoutParameters
+
+class FirstStartWithNoNetworkActivity : TypefaceStyleableActivity() {
+
+    private lateinit var continuationIntent: Intent
+
+    private var networkCallback = NetworkConnectedCallback()
+    private var mainThreadHandler = Handler(Looper.getMainLooper())
+
+    private lateinit var connectivityManager: ConnectivityManager
+
+    private var shortAnimationDuration: Int = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_first_start_with_no_network)
+
+        DialogLayoutParameters.wrapHeight(this)
+                .applyTo(window)
+
+        firstStartNevermind.setOnClickListener { finish() }
+
+        continuationIntent = intent.getParcelableExtra(EXTRA_CONTINUATION_INTENT)
+
+        connectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        shortAnimationDuration = resources.getInteger(android.R.integer.config_shortAnimTime)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        val networkRequest = NetworkRequest.Builder()
+                .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                .build()
+        connectivityManager.registerNetworkCallback(networkRequest, networkCallback)
+
+        doIfAnimatedVectorDrawable(firstStartProgress.drawable) { it.start() }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        connectivityManager.unregisterNetworkCallback(networkCallback)
+        doIfAnimatedVectorDrawable(firstStartProgress.drawable) { it.stop() }
+    }
+
+    private fun doIfAnimatedVectorDrawable(drawable: Drawable, action: (AnimatedVectorDrawable) -> Unit) {
```

Comment:
usually these read nicer if you have `action: AnimatedVectorDrawable.() -> Unit`, then you can call it as `doIfAnimatedVectorDrawable(drawable) { stop() }`

not a blocker

Reply:
fd60dc074296f41dd74b5c8d6cfe3b8ccf3f7616